sdk/core/Azure.Core.Experimental/src/JsonPatchDocument.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Azure.Core.JsonPatch;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents a JSON Patch document.
+    /// </summary>
+    public class JsonPatchDocument
+    {
+        internal Collection<JsonPatchOperation> Operations { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="JsonPatchDocument"/>
+        /// </summary>
+        public JsonPatchDocument()
+        {
+            Operations = new Collection<JsonPatchOperation>();
+        }
+
+        /// <summary>
+        /// Appends an "add" operation to this <see cref="JsonPatchDocument"/>.
+        /// </summary>
+        /// <param name="path">The path to apply the addition to.</param>
+        /// <param name="rawJsonValue">The raw JSON value to add to the path.</param>
+        public void AppendAdd(string path, string rawJsonValue)
+        {
+            Operations.Add(new JsonPatchOperation(JsonPatchOperationKind.Add, path, null, rawJsonValue));
+        }
+
+        /// <summary>
+        /// Appends a "replace" operation to this <see cref="JsonPatchDocument"/>.
+        /// </summary>
+        /// <param name="path">The path to replace.</param>
+        /// <param name="rawJsonValue">The raw JSON value to replace with.</param>
+        public void AppendReplace(string path, string rawJsonValue)
+        {
+            Operations.Add(new JsonPatchOperation(JsonPatchOperationKind.Replace, path, null, rawJsonValue));
+        }
+
+        /// <summary>
+        /// Appends a "copy" operation to this <see cref="JsonPatchDocument"/>.
+        /// </summary>
+        /// <param name="from">The path to copy from.</param>
+        /// <param name="path">The path to copy to.</param>
```

Comment:
Should this be named "to" or "fromPath"/"toPath"

Reply:
I'm trying to keep the names similar between parameter and the resulting json.